### PR TITLE
class name resolution

### PR DESCRIPTION
**Pull request by amberovsky**
*Description of changes:*
Use ::class for class name resolution 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by stobrien89 — APPROVED** [review]

**Comment by stobrien89** [comment]
Thanks for the contribution, @amberovsky!

**Review by stobrien89 — CHANGES_REQUESTED** [review]
@amberovsky,

Thanks for all of the work on this.  I just had a few things we'd like to change.  Otherwise, very appreciative of the effort— helps us to be a bit more consistent and idiomatic.

File: src/ClientResolver.php [hunk 1]
```diff
@@ -548,7 +549,7 @@ public static function _apply_credentials($value, array &$args)
             $args['credentials'] = CredentialProvider::defaultProvider($args);
         } else {
             throw new IAE('Credentials must be an instance of '
-                . 'Aws\Credentials\CredentialsInterface, an associative '
```
[the remark this answers is not in the document]
Comment (amberovsky) [reply src/ClientResolver.php:551]: 🤦

[the remark this answers is not in the document]
Comment (amberovsky) [reply src/ClientResolver.php:551]: Will fix it, I think it should be put in the backlog to write a test on this code block

**Review by stobrien89 — APPROVED** [review]

**Comment by stobrien89** [comment]
Thanks for the contribution @amberovsky! And apologies for the delay in getting back to this.

**Pull request merged**